website/blog/2024-02-07-bismuth.md

```diff
@@ -0,0 +1,480 @@
+  - improvement: Discover sbt in path before fallbacking to launcher. [kasiaMarek](https://github.com/kasiaMarek)
+  - improvement: Only save scala or java fingerprints. [tgodzik](https://github.com/tgodzik)
+  - bugfix: Offer auto-import for new classes created via file provider. [kasiaMarek](https://github.com/kasiaMarek)
+  - improvement: Add support for tests using weaver cats effects. [kasiaMarek](https://github.com/kasiaMarek)
+  - bugfix: Show `run`/`debug` code lenses when file contains shebang header. [kasiaMarek](https://github.com/kasiaMarek)
+  - bugfix: Make debug completions independent of the order of stackTrace being requested.
+  [mekpavit](https://github.com/mekpavit)
+  - bugfix: Fix go to definition for toplevel definitions in files with `.` in the name. [kasiaMarek](https://github.com/kasiaMarek)
+  - improvement: Add `case` keyword completion. [jkciesluk](https://github.com/jkciesluk)
+  - bugfix: Don't tokenize if the source hasn't changed. [tgodzik](https://github.com/tgodzik)
+  - bugfix: Handle missing line argument in debug completion request. [mekpavit](https://github.com/mekpavit)
+  - feature: Cancel worksheet evaluations when canceling compilation. [ckipp01](https://github.com/ckipp01)
+  - feature: Show import renames in hover. [kasiaMarek](https://github.com/kasiaMarek)
+  - [Scala 2] bugfix: Document highlight in self type. [jkciesluk](https://github.com/jkciesluk)
+  - improvement: Remove semantic tokens fallback. [jkciesluk](https://github.com/jkciesluk)
+  - bugfix: Handle missing jar file in SetBreakpointRequest. [adpi2](https://github.com/adpi2)
+  - bugfix: Don't show default argument values if type is Any. [tgodzik](https://github.com/tgodzik)
+  - [Scala 3] bugfix: Don't show givens as completions from classpath search. [kasiaMarek](https://github.com/kasiaMarek)
+  - [Scala 3] bugfix: Don't show scope completions inside multi-imports. [kasiaMarek](https://github.com/kasiaMarek)
+  - improvement: Show workspace completions for short queries. [kasiaMarek](https://github.com/kasiaMarek)
+  - improvement: Add atomic update on TrieMap. [jkciesluk](https://github.com/jkciesluk)
+  - bugfix: Fix StacktraceAnalyser in case of empty package. [tgodzik](https://github.com/tgodzik)
+  - [Scala 3] bugfix: Wrong span in extension methods in context functions. [jkciesluk](https://github.com/jkciesluk)
+  - bugfix: Fix searching for implementations of an abstract type. [tgodzik](https://github.com/tgodzik)
+  - bugfix: Fix java support (hover/completions) for symbols from dependencies. [tgodzik](https://github.com/tgodzik)
+  - [Scala 3] improvement: Suggest workspace completions for implicit classes. [tgodzik](https://github.com/tgodzik)
+  - bugfix: Extract method with empty lines. [kasiaMarek](https://github.com/kasiaMarek)
+  - feature: Add option to run verbose compilation. [tgodzik](https://github.com/tgodzik)
+  - improvement: Fetch missing dependency sources. [kasiaMarek](https://github.com/kasiaMarek)
+
+
```

Comment:
Do you think it's worth mentioning adding "progress"? I know for a lot of clients (especially those that don't use metals status), this is a really big feature. You could specifically mention that for clients that don't support metals status this is available. One other thing to mention though is that by default neither metals status nor progress is turned on. I found this a bit confusing and I was actually wondering why it wasn't working when I turned off metals status. It needs to explicitly be set to "off" instead of the default.

Reply:
I'm surprised by this, when I tested on Helix progress "just worked" without needing any configuration?

Reply:
Ahhh, you're actually correct. I realize now that in nvim-metals when I don't overwrite it with "off" it sends in "show-message". So ignore that point it's incorrect.

Reply:
worth adding it to the TLDR as well?